src/main/java/net/sf/mzmine/modules/peaklistmethods/identification/sirius/table/PreviewCell.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2006-2018 The MZmine 2 Development Team
+ *
+ * This file is part of MZmine 2.
+ *
+ * MZmine 2 is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 2 of the License, or (at your option) any later
+ * version.
+ *
+ * MZmine 2 is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+ * A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * MZmine 2; if not, write to the Free Software Foundation, Inc., 51 Franklin St,
+ * Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+package net.sf.mzmine.modules.peaklistmethods.identification.sirius.table;
+
+import java.awt.Color;
+import java.awt.Font;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.Rectangle;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.swing.JComponent;
+
+import org.openscience.cdk.interfaces.IAtomContainer;
+import org.openscience.cdk.renderer.AtomContainerRenderer;
+import org.openscience.cdk.renderer.RendererModel;
+import org.openscience.cdk.renderer.color.CDK2DAtomColors;
+import org.openscience.cdk.renderer.font.AWTFontManager;
+import org.openscience.cdk.renderer.generators.BasicSceneGenerator;
+import org.openscience.cdk.renderer.generators.IGenerator;
+import org.openscience.cdk.renderer.generators.standard.StandardGenerator;
+import org.openscience.cdk.renderer.visitor.AWTDrawVisitor;
+
+/**
+ * PreviewCell for chemical structure images
+ */
+public class PreviewCell extends JComponent {

```

Comment:
The more common strategy is to make PreviewCell also implement TableCellRenderer, and the getTableCellRendererComponent will just return 'this'. That way the component gets reused instead of being generated again each time the cell is visible. See http://www.javapractices.com/topic/TopicAction.do?Id=168

Reply:
I decided that it will be too complicated and splitted into to two classes.

Reply:
It is not possible to do in one class in this case without cranches, I reimplemented a solution with `new` objects only during their initialization.

Reply:
Why is it too complicated? Like I said, it is a common practice.